### PR TITLE
feat: add input validation module (Issue #10)

### LINKED ISSUE #10: [Security] Add Input Validation Module
## Goal
Create input validation module for database/user identifiers to prevent SQL injection.

## Description
As part of Week 1 security fixes, we need to add a Validators module that validates and sanitizes database names, usernames, and file paths. This module will be used by all command classes to ensure input safety before executing SQL queries or shell commands.

## Files
- Create `lib/pgchief/validators.rb`
- Create `spec/pgchief/validators_spec.rb`
- Modify `lib/pgchief.rb` to require validators

## Acceptance Criteria
- [ ] Module provides `valid_identifier?` method that validates PostgreSQL naming rules (starts with letter/underscore, alphanumeric/underscore only, max 63 chars)
- [ ] Module provides `sanitize_identifier` that raises `Pgchief::InvalidIdentifierError` for invalid input
- [ ] Module provides `valid_password?` and `sanitize_password` for password validation (length, non-empty)
- [ ] Module provides `valid_file_path?` and `sanitize_file_path` for file path validation (prevents traversal, shell metacharacters)
- [ ] Comprehensive test coverage for edge cases (SQL injection attempts, special characters, empty/nil values)
- [ ] All existing tests pass after integrating the module

## Reference
See `docs/plans/2026-01-30-week1-fix-sql-shell-injection.md` Task 1 for detailed implementation steps.

## Priority
High (security fix)

### PR DESCRIPTION
**Pull request by bully-goose**
Implements input validation module as per Week 1 security plan.

- Adds  module with , , , , , .
- Adds  and  exception classes.
- Includes comprehensive test suite.
- Required by subsequent security fixes.

Closes #10.

### CODE REVIEW

File: lib/pgchief/validators.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Pgchief
+  class InvalidIdentifierError < StandardError; end
+  class InvalidFilePathError < StandardError; end
```
Comment (jayroh) [inline lib/pgchief/validators.rb]: Move these 2 error classes into the `Pgchief::Errors` namespace in `lib/pgchief.rb` and adjust this class to reflect that

**Review by jayroh — COMMENTED** [review]
Just one comment on a fix
